eth/filters/filter_system.go

```diff
@@ -68,6 +72,8 @@ const (
 	logsChanSize = 10
 	// chainEvChanSize is the size of channel listening to ChainEvent.
 	chainEvChanSize = 10
+	// stateEvChanSize is the size of channel listening to ChainEvent.
+	stateEvChanSize = 10
```

Comment:
Are you sure the number of messages to this channel won't exceed 10? Because if they do there will be data loss.

Reply:
Won't be data loss, just stuck there till the previous 10 messages get broadcasted. @ptsayli please test and check.

Reply:
Yeah that's what I meant, you won't be able to push to channel hence the value will be discarded.

Reply:
Tested
Working Fine

Reply:
Yeah, in `stateCh:  make(chan core.NewStateChangeEvent, stateEvChanSize)` , `stateEvChanSize` specifies the number of elements that will be buffered in the channel, before it starts to block on events being consumed. So this should work.